res_partner_operating_unit/hooks.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2020 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+
+def pre_init_hook(cr):
+    # Add new table and columns to hold values
+    cr.execute("""
+        CREATE TABLE operating_unit_partner_rel (
+            partner_id INTEGER NOT NULL
+                REFERENCES res_partner(id) ON DELETE CASCADE,
+            operating_unit_id INTEGER NOT NULL
+                REFERENCES operating_unit(id) ON DELETE CASCADE,
+        );
+    """)
+    # Add the values
+    cr.execute("""
+        INSERT INTO operating_unit_partner_rel
+            (partner_id, operating_unit_id)
+        SELECT id, 1 FROM res_partner;
```

Comment:
Why operating_unit is always the first?

Reply:
I don't know...

Reply:
I am not too familiar with the module. However, if we add partners to all OUs on the installation, all partners will belong to all OU, won't they? when the module is not installed all partners belong to all OU. From my point of view, if we include by default all partners to all OUs will be consistent. So I'd loop over all OUs assigning partners. 

Does that look correct to you?

Reply:
Humm not really. The field is required so we need to set a default value. The only value we have is the "Main OU" created by the `operating_unit` module.

The hook and the migration script only set one OU ("Main OU") to all the partners (hooks) or all the partners with no OU (migration script).

Reply:
hmm I see. Ok from my side go ahead with this. The only doubt is if there is a chance that id=1 was delete from some reason before the module is installed. Perhaps we can do a select before.

Reply:
@max3903  Why are the migration script and hook needed?
Isn't the default method in operating_unit_ids working and assigning the user default_operating_unit_id?

Reply:
@BT-rmartin Defaults works when you create a new record. It will not update the existing ones. 

The hook is executed when the module is installed and set the Main OU on all existing partners.

The migration script is executed when the module is upgraded and set the Main OU on existing partners with no OU.

Reply:
@max3903 I did check and defaults are applied also to existing data. Maybe not m2m, but for basic fields they are. 
From my understanding, default is applied to both new and existing data. 
pre-init/post-init hooks only when installing for existing data.
migration script only when upgrading for existing data. 
In fact I see in another PR https://github.com/OCA/operating-unit/pull/225 that you removed the hook. Is this really necessary here?